### PR TITLE
Remove satchel gathering limit of 15.

### PR DESCRIPTION
**Pull request by WrillWasTaken**
## About The Pull Request
Makes it so satchels will pick up as many items as they can fit, instead of up to 15. 

## Why It's Good For The Game
Clicking 5-7 times per-tile to gather ore isn't fun for miners. I don't see why other jobs would want to have to mash their left click just to pick up stuff from a tile either.

## Testing
I tried picking up a pile of plasma with the ore satchel. More than 15 items got picked up. 

## Changelog
:cl: Wrill
tweak: satchels will gather as many items as they can store, instead of up to 15.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/storage/storage.dm [hunk 1]
```diff
@@ -22,6 +22,7 @@
 	var/collection_mode = TRUE //0 = pick one at a time, 1 = pick all on tile
 	var/use_sound = "rustle" //sound played when used. null for no sound.
 	var/is_tray_hidden = FALSE //hides from even t-rays
+	var/pickup_amount = 0 //the amount of stuff(tm) picked up when gathering items from a tile. limit is ignored if this is 0
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/storage/storage.dm:25]:
```suggestion
```

File: code/game/objects/items/weapons/storage/storage.dm [hunk 2]
```diff
@@ -473,10 +474,14 @@
 /obj/item/storage/proc/collectItems(turf/target, mob/user)
 	ASSERT(istype(target))
 	. = FALSE
-	var/limiter = 15
+	var/limiter
+	if(pickup_amount > 0)
+		limiter = pickup_amount
+
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/storage/storage.dm:480]:
```suggestion
```

File: code/game/objects/items/weapons/storage/storage.dm [hunk 3]
```diff
@@ -473,10 +474,14 @@
...
 	for(var/obj/item/I in target)
-		if(--limiter < 0)
-			break
+		if(pickup_amount > 0)
+			if(--limiter < 0)
+				break
```
Comment (SirRichardFrancis) [inline code/game/objects/items/weapons/storage/storage.dm:484]:
```suggestion
```

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]
Good PR, but while you're at it just remove this mechanic altogether, we don't need it.

**Comment by WrillWasTaken** [comment]
my original thought process was that I wasn't sure if someone DID need it. if nobody does then good riddance to that awful mechanic.

**Review by SirRichardFrancis — APPROVED** [review]

**Review by Humonitarian — APPROVED** [review]

**Pull request merged**